assets/js/custom/ProgramCredits.js

```diff
@@ -1,50 +1,196 @@
 import $ from 'jquery'
 /* global Routing */
 
-export function ProgramCredits (programId) {
+export function ProgramCredits (programId, translationSaved, translationDeleted, creditsSelect,
+  closeEditorDialog, keepOrDiscardDialog, snackbarNotification, customTranslationApi) {
   const credits = $('#credits')
   const editCreditsUI = $('#edit-credits-ui')
   const editCredits = $('#edit-credits')
+  const editCreditsButton = $('#edit-credits-button')
+  const closeCreditsEditorButton = $('#close-credits-editor-button')
+  const creditLanguageSelectorList = $('#credit-language-selector-list')
+  const creditsSelectedText = $('#credits-selected-text')
+  const editCreditsSubmitButton = $('#edit-credits-submit-button')
   const editCreditsError = $('#edit-credits-error')
   const descriptionCreditsContainer = $('#description-credits-container')
   const showMoreToggle = $('#descriptionShowMoreToggle')
   const descriptionHeadline = $('#description-headline')
 
-  $('.edit-credits-button').on('click', () => {
-    if (editCreditsUI.hasClass('d-none')) {
-      descriptionCreditsContainer.hide()
-      descriptionHeadline.hide()
-      editCreditsUI.removeClass('d-none')
-      showMoreToggle.addClass('d-none')
-    } else {
-      descriptionCreditsContainer.show()
-      descriptionHeadline.show()
-      editCreditsUI.addClass('d-none')
-      handleShowMore()
-    }
+  let languages = {}
+  let previousCreditsIndex = 0
+  let lastSavedCredits = credits.text().trim()
+
+  $(document).ready(function () {
+    getLanguages()
   })
 
-  $('#edit-credits-submit-button').on('click', () => {
+  function getLanguages () {
+    $.ajax({
+      url: '../languages',
```

Comment:
I think I can create a class that will get the languages, then instantiate it in the program.js webpack file and pass it into the constructors for ProgramDescription and ProgramCredits. Or I can just add it to the CustomTranslationApi class?

Reply:
but then program.js and translation.js still separated fetch the same api. 

Is it possible with the current architecture to have a singleton like described here? https://www.digitalocean.com/community/tutorials/js-js-singletons

Reply:
So, I was able to create a static class that you can import into any js file. Although, there are still multiple API calls being made because the ajax call on my local is really slow, not sure if this will be fixed on prod or if there will still be multiple files. 

Do you have any ideas on how I can reduce the API calls within this structure? I tried a couple of different types of locks and looked into some async functions, but nothing seemed to work really well without some sort of polling. Or are you okay with the class I just pushed?

Reply:
i don't understand how locking is needed, i though javascript is not multithreaded. I might run it later

Reply:
It ends up calling the ajax function multiple times because the ajax is asynchronous and takes a while to complete and set the languages variable. I've looked into it a bit and done some testing, but there doesn't seem to be a clean solution

Reply:
ok can we revert this commit and deal with it later?

Reply:
What about calling the getLanguages API when the user opens the editor since it is not used any other time? There might be a short delay the first time they open it and select a language, but when they open the credits there shouldn't be another API call

Reply:
It's called when the page loads in Translation.js and subsequent calls will return 304 and use cached value

Reply:
Okay, sounds good. Do you want me to just move the ajax call to the CustomTranslationApi or leave it in its own class or should I just revert it and we will deal with it later as you said?

Reply:
get drop this commit for now 437f010edea44f9a66f8487cbca15869eb831c9e

Reply:
I reverted the commit